lib/ansible/executor/task_executor.py

```diff
@@ -211,8 +211,8 @@ def _get_loop_items(self):
         if self._task.loop_with:
             if self._task.loop_with in self._shared_loader_obj.lookup_loader:
                 fail = True
-                if self._task.loop_with == 'first_found':
-                    # first_found loops are special. If the item is undefined then we want to fall through to the next value rather than failing.
+                if self._task.loop_with == 'first_found' or self._task.loop_with == 'file_found':
+                    # first_found and file_found loops are special. If the item is undefined then we want to fall through to the next value rather than failing.
```

Comment:
lookups have a toggle now to deal with errors, this is something we want to remove, they should not be hard coded into task executor.